core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/CurateOutcomeBuildItem.java

```diff
@@ -14,4 +20,39 @@ public CurateOutcomeBuildItem(ApplicationModel appModel) {
     public ApplicationModel getApplicationModel() {
         return appModel;
     }
+
+    /**
+     * Returns the application module directory, if the application is built from a source project.
+     * For a single module project it will the project directory. For a multimodule project,
+     * it will the directory of the application module.
+     * <p>
+     * During re-augmentation of applications packaged as {@code mutable-jar} this method will return the current directory,
+     * since the source project might not be available anymore.
+     *
+     * @return application module directory, never null
+     */
+    public Path getApplicationModuleDirectory() {
```

Comment:
if it doesn't return null, we need a way to know if it exists because there is some logic (such as using an absolute directory for Quinoa as a fallback: https://github.com/quarkiverse/quarkus-quinoa/blob/main/deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java#L404)

Reply:
Sure, let's clarify what you are going to be using the module directory for. I guess the module dir itself won't be the target directory you'll be using?
When you say `we need a way to know if it exists`, what do you mean by `it`?